src/arangodb.c

```diff
@@ -0,0 +1,1201 @@
+
+    if ( 0 == ret_val ) {
+      ret_val = curl_perform(&service->curlinfo, &service->endpoint, "/_admin/statistics-description", &desc_node);
+    } // if
+
+    if ( 0 == ret_val ) {
+
+      if (NULL != service_node && NULL != desc_node) {
+        service_decode_stats(service, service_node, desc_node);
+      } // if
+    } // if
+    
+    yajl_tree_free(service_node);
+    yajl_tree_free(desc_node);
+
+    // rocksdb statistics retrieved every pass if engine
+    if (NULL != service->engine && 0 == strcasecmp("rocksdb", service->engine)) {
+      ret_rocks = service_get_rocksdb(service);
+    } // if
+
+    // possible for basic stats to be disabled and rocksdb to be enabled
+    //  or vice-versa ... if one is good, show happiness
+    if (0 == ret_val || 0 == ret_rocks) {
+      ret_val = 0;
+    } // if
+  } else {
+    ERROR("arangodb plugin: service_read: Invalid user data.");
+    ret_val = -1;
+  } // else
+  
+  return ret_val;
+
+} // service_read
+
+
+static void service_decode_stats(service_t * service, yajl_val srv_stats, yajl_val stats_desc) {
+  cdtime_t now;
+  size_t idx;
+  yajl_val figures_array, group, identifier, type, stat, object;
+  const char * figures_path[] = { "figures", (const char *) 0 };
+  const char * group_path[] = { "group", (const char *) 0 };
+  const char * identifier_path[] = { "identifier", (const char *) 0 };
+  const char * type_path[] = { "type", (const char *) 0 };
+  char * stats_path[] = { NULL, NULL, (char *) 0, (char *) 0  };
+
+  now = cdtime();
+  figures_array = yajl_tree_get(stats_desc, figures_path, yajl_t_array);
+  if (YAJL_IS_ARRAY(figures_array)) {
+    service_data_set_init(figures_array);
+
+    for (object=figures_array->u.array.values[0], idx=0;
+         idx<figures_array->u.array.len;
+         ++idx, object=figures_array->u.array.values[idx]) {
+      stat = NULL;
+      group = yajl_tree_get(object, group_path, yajl_t_string);
+      identifier = yajl_tree_get(object, identifier_path, yajl_t_string);
+      type = yajl_tree_get(object, type_path, yajl_t_string);
+
+      if (YAJL_IS_STRING(group) && YAJL_IS_STRING(identifier) && YAJL_IS_STRING(type)) {
+        stats_path[0] = group->u.string;
+        stats_path[1] = identifier->u.string;
+        stats_path[2] = (char *)0;
+        
+        stat = yajl_tree_get(srv_stats, (const char **)stats_path, yajl_t_any);
+        if (YAJL_IS_OBJECT(stat)) {
+          stats_path[2] = "sum";
+          stat = yajl_tree_get(srv_stats, (const char **)stats_path, yajl_t_number);
+        } // if
+      } // if
+      
+      if (NULL != stat && YAJL_IS_NUMBER(stat)) {
+        value_t value;
+        value_list_t vl = VALUE_LIST_INIT;
+        bool skip = false;
+        int ret_val;
+        
+        vl.values = &value;
+        vl.values_len = 1;
+        vl.time = now;
+        // default interval?
+        ssnprintf(vl.host, DATA_MAX_NAME_LEN, "%s", service->endpoint.host);
```

Comment:
Don't use DATA_MAX_NAME_LEN.
Please use `sizeof()` (sizeof(vl.host) and so on).